websocket/client.go

```diff
@@ -58,24 +70,41 @@ type Client struct {
 	send chan Message
 }
 
-func NewClient(userID string, topics []TopicDescriptor, conn *websocket.Conn, send chan Message) *Client {
+func NewClient(userID string, topics []TopicDescriptor, conn *websocket.Conn, send chan Message) Client {
 	// topicsMap := make(map[TopicDescriptor]struct{})
 	// for _, topicDescriptor := range topics {
 	// 	topicsMap[topicDescriptor] = struct{}{}
 	// }
 	uuid := uuid.New()
 	sessionID := uuid.String()
-	return &Client{
+	return &client{
 		sessionID: sessionID,
-		UserID:    userID,
-		Topics:    topics,
+		userID:    userID,
+		topics:    topics,
 		conn:      conn,
 		send:      send,
 	}
 }
 
-func (client *Client) SessionID() string {
-	return client.sessionID
+func (c *client) SessionID() string {
+	return c.sessionID
+}
+
+func (c *client) UserID() string {
+	return c.userID
+}
+
+func (c *client) Topics() []TopicDescriptor {
+	return c.topics
+}
+
+func (c *client) SendMessage(m Message) bool {
```

Comment:
Hide the internal implementation of how messages are consumed by the client.

Reply:
Still figuring out how to get hub works on mobile here. Can we do a context time out here in case the channel is temporarily blocked?

Reply:
Commented on a change that needs to happen in this method

Reply:
I was following the same pattern that was being done in the select statement in the server here: https://github.com/atljoseph/go-websocket-pusher-example-api/blob/a05f317477c9d5075ec0f346addacacc0712a331/websocket/server.go#L328.

I'll let you decide whether it needs to be changed. The main goal of this change was to remove the ability for `Server` to send messages to the client's `send` channel directly . `Server` should not care how `Client` is sending its messages.